File: 019-count-sunday.js
function isLeapYear(year) {
  if (year % 400 === 0) return true;
  if (year % 100 === 0) return false;
  if (year % 4 === 0) return true;
  return false;
}

const daysInMonth = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];

let dayOfWeek = 2; 
let count = 0;

for (let year = 1901; year <= 2000; year++) {
  for (let month = 0; month < 12; month++) {
    if (dayOfWeek === 0) {
      count++;
    }

    let days = daysInMonth[month];
    if (month === 1 && isLeapYear(year)) {
      days = 29;
    }

    dayOfWeek = (dayOfWeek + days) % 7;
  }
}

console.log(count);

File: 019-count-sunday.py
def is_leap_year(year):
    if year % 400 == 0:
        return True
    if year % 100 == 0:
        return False
    if year % 4 == 0:
        return True
    return False

def count_sunday():
    
    days_in_month = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]

    if (is_leap_year):
        day = 366 % 7
    else:
        day = 365 % 7
    
    day_of_week = day +1 
    count = 0

    for year in range(1901, 2001):
        for month in range(12):
            if day_of_week == 0: 
                count += 1
        
            days = days_in_month[month]
            if month == 1 and is_leap_year(year):
                days = 29
            day_of_week = (day_of_week + days) % 7

    return count
    
print(count_sunday())

